PersonBook/PersonBook/PersonBook/Service/PersonService.cs

```diff
@@ -0,0 +1,68 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PersonBook.Service
+{
+    using CustomExceptions;
+    using Entity;
+
+    public class PersonService : IPersonService
+    {
+        private ICollection<Person> _personList;
+        public PersonService()
+        {
+            _personList = new List<Person>();
+        }
+
+        public void AddPerson(string name, string surname)
+        {
+            int Id = IdGenerator();
+
+            _personList.Add(
+                   new Person
+                   {
+                       Id = Id,
+                       FirstName = name,
+                       LastName = surname
+                   });
+        }
+
+        public void DeletePerson(int id)
+        {
+
+            var personToRemove = _personList.SingleOrDefault(p => p.Id == id);
```

Comment:
SingleOrDefault is not a good solution here. You could user FirstOrDefault instead and then check if it's null. if it's null you can let the user know the issue and then return;

Reply:
variable name could be just `person`

Reply:
Done